File: Python3/601 - 800/643 Maximum Average Subarray I.py
# 896ms, 81.33%; 16.2MB, 12.50%
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        _sum = _max = sum(nums[:k])
        for i in range(k, len(nums)):
            _sum += nums[i] - nums[i - k]
            _max = max(_max, _sum)
        return _max / k

File: 601 - 800/643 Maximum Average Subarray I.cc
// 176ms, 67.29%; 16.8MB, 33.33%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    double findMaxAverage(const vector<int> &nums, int k) {
        int _sum = accumulate(nums.begin(), nums.begin() + k, 0), _max = _sum;
        for (int i = k; i < nums.size(); ++i) {
            _sum += nums[i] - nums[i - k];
            _max = max(_max, _sum);
        }
        return _max / double(k);
    }
};
